src/Kiota.Builder/EqualityComparers/GlobComparer.cs

```diff
@@ -9,11 +9,14 @@ internal class GlobComparer : IEqualityComparer<Glob>
 {
     public bool Equals(Glob? x, Glob? y)
     {
-        return x == null && y == null || x != null && y != null && GetHashCode(x) == GetHashCode(y);
+        if (x is null || y is null) return object.Equals(x, y);
+        return string.Equals(x.ToString(), y.ToString(), StringComparison.OrdinalIgnoreCase);
```

Comment:
Why not simply call get hash code to reduce duplication further?

Reply:
Because of [hash collisions](https://en.wikipedia.org/wiki/Hash_collision). The contract of Equals MUST return false for different values, no?

Reply:
Hash collisions are a problem only if the "represented" information is different but results in the same hash code. In which case we should review the implementation of GetHashCode because it's going to cause other side effects (building indexed collections, etc...)
Effectively objects are trees of data. If we have two trees with the same structure (same number of nodes, edges placed in the same way etc...) and the same values (leaves contain the same values) they should be considered equal even if they come from different memory spaces (and their references are different).
I'm not sure how having diverging implementations for the indexed collections scenario and for the comparison scenario is going to help? which problem it'll solve?
Additionally, having diverging implementations will require additional maintenance.

Reply:
Also, I think you'll have bugs in collections as well if your comparer equality depends on GetHashCode. If you are adding an element to a Set or a Map, the hash code will determine if there *might* be an equal element. For example, assume item A and B are strings with the same hash code, but different values. If you try to add both to a set, then they'll be added to the same slot. If you depend on hash code for equality checks, then B will not be added to the set and you'll lose information.

Reply:
The point I'm trying to make is that I'm not sure why we'd care about potential collisions in one scenario but not another?

Reply:
We don't care about collisions in hash code implementation because avoiding collisions in the hash code is impossible. We care about collisions in the Equals because 2 values being equal means they are the same. When a collision occurs, 2 values that are different will have the same hash code.

Reply:
ok the trick here is that indexed collections are designed to handle this concern.
If a collision happens with GetHashCode, it'll then call Equals to know whether the objects are truly equals and what it should do.
http://dotnetfiddle.net/RD5195

This effectively addresses my concern, thanks for the discussion on the topic. Resolving.